crates/core/src/telemetry.rs

```diff
@@ -133,7 +153,11 @@ pub fn log(level: &str, target: &str, kv: &[(&str, Option<ValueBag>)]) {
 				match v {
 					None => {},
 					Some(i) => {
-						s.serialize_entry(k, i)?;
+						if is_sensitive_field(k) {
```

Comment:
Sorry if I was misleading on the slack thread.

If a user is getting here, they explicitly added a logging field like `token: request.headers["token"]`. So they _asked_ for this (possibly sensitive) header to be logged; we should respect it. 

Note: this log line is called _only_ for the access logs, not all logs. But even so, we are _probably_ never setting a key like `token` and expecting it to not log, since then we explicitly wrote code to log it. More likely accidental encoding would be encoding it inside a string value like `info!("{request:?})"`

Reply:
Ah you have the same code on the non-accesslog path. But same comment

Reply:
Yep, updated this- I think what we want is a user-configured option to redact logs? So by default if they asked the header to be logged, it will, unless it's in the redact headers set. So the sensitive_headers logic hasn't changed except for checking if the header is in the redacted list.